### PR TITLE
Add a script to build a standalone binary

### PR DESCRIPTION
**Pull request by agateau-gg**
This PR adds the `scripts/build-standalone-exe` script. This script builds a standalone ggshield executable using [Pyinstaller](https://pyinstaller.org/en/stable/).

It builds binaries for:
- Windows x86_64
- Linux x86_64
- Mac x86_64
- Mac M1

The binaries can be downloaded from the bottom of the [summary page](https://github.com/GitGuardian/ggshield/actions/runs/7888013682?pr=833) of the action run (after all the deprecation warnings 😅)

Note: Building for Mac M1 required bumping the default Python version to 3.10 because 3.9 was not available on the GitHub M1 runner.

Best reviewed commit-by-commit.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -153,6 +153,71 @@ jobs:
...
+
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install normal dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade pipenv==2023.10.3
```
Comment (fnareoh) [inline .github/workflows/ci.yml:181]: Should we consider moving shield to pdm at some point ?

**Comment by fnareoh** [comment]
Looks good to me !

**Review by fnareoh — APPROVED** [review]
Looks good to me !

File: .github/workflows/ci.yml (hunk 1 above)
Comment (agateau-gg) [reply .github/workflows/ci.yml:181]: Yes. If I am not mistaken, @deronnax plans to work on it at some point.

Comment (deronnax) [reply .github/workflows/ci.yml:181]: yup

**Pull request merged**